barretenberg/cpp/src/barretenberg/translator_vm/translator_proving_key.cpp

```diff
@@ -52,6 +52,41 @@ void TranslatorProvingKey::compute_concatenated_polynomials()
     parallel_for(concatenation_groups.size() * concatenation_groups[0].size(), ordering_function);
 }
 
+/**
+ * @brief Sequential method for computing the concatenated polynomials by interleaving.
+ *
+ * @note This is not currently used in the implementation.
+ */
+void TranslatorProvingKey::compute_concatenated_polynomials_by_interleaving()
+{
+    auto groups = proving_key->polynomials.get_groups_to_be_concatenated();
+    auto concatenated_polynomials = proving_key->polynomials.get_concatenated();
+
+    // Targets have to be full-sized proving_key->polynomials. We can compute the mini circuit size from them by
+    // dividing by concatenation index
+    const size_t MINI_CIRCUIT_SIZE = concatenated_polynomials[0].size() / Flavor::CONCATENATION_GROUP_SIZE;
+    ASSERT(MINI_CIRCUIT_SIZE * Flavor::CONCATENATION_GROUP_SIZE == concatenated_polynomials[0].size());
+
+    for (auto [concatenated, group] : zip_view(concatenated_polynomials, groups)) {
+        interleave(group, concatenated);
+    }
+}
+
+/**
+ * @brief Construct a concatenated polynomial from a group of polynomials by interleaving.
+ */
+void TranslatorProvingKey::interleave(const RefVector<Polynomial>& group, Polynomial& result)
+{
+
+    const size_t group_size = group.size();
+    const size_t group_polynomial_size = result.size() / group_size;
```

Comment:
Its a little hard to follow how these sizes work out. `group_polynomial_size` is computed as the the `size` (size of the memory block) of the destination poly divided by number of small polys, but then we only iterate from `result.start_index()` to `group_polynomial_size`. If `result` had a start index > 0, wouldn't this not iterate over the full `size`? Perhaps some checks within this method would add safety and also clarify what the sizes are expected to be?

Reply:
size() is computed as end_index() - start_index()

Reply:
maybe we're saying the same thing but just to be sure: I agree size is end - start, but the loop below goes from idx = start to idx = size, not idx = end, i.e. it does not loop over size-many elements. Oh but group_polynomial_size is not being computed `group_polynomial.size()` so maybe its somehow doing the right thing?

Reply:
yeah so the issue here is that the group polynomials are initialised over the full domain (i.e. the same domain as result) rather than they're real domain (mini_circuit_size) even though past their real domain the coeff are 0. A follow on is to refine that so we can call group.size(). I started doing it in this PR but subtle changes to sizes deserve a PR of their own so I added an issue for now.